include/sharg/validators.hpp

```diff
@@ -938,13 +937,16 @@ namespace detail
  *
  * \remark For a complete overview, take a look at \ref argument_parser
  */
-template <typename option_value_t>
 struct default_validator
 {
-    //!\brief Type of values that are tested by validator
-    using option_value_type = option_value_t;
+    //!\brief Dummy type needed for member sharg::detail::default_validator::option_value_type.
+    struct empty_type {};
+
+    //!\brief Dummy option_value_type to model sharg::validator but any type is accepted in the `operator()`.
+    using option_value_type = empty_type;
```

Comment:
Because it is a typedef, and not a member, it doesn't matter whether the class is empty or not (it would matter if we stored something of type `option_value_type`).
We only need `option_value_type` for the concept, so `option_value_type` could be any type.... why not `std::any` 😁 

```suggestion
    //!\brief Dummy option_value_type to model sharg::validator but any type is accepted in the `operator()`.
    using option_value_type = std::any;
```

Would need an `#include <any>`

Doesn't really make a difference, but `any` might get the message better across than `empty`

Reply:
I like that :)